src/server/rdb_load.cc

```diff
@@ -1494,12 +1509,18 @@ auto RdbLoaderBase::ReadIntSet() -> io::Result<OpaqueObj> {
 }
 
 auto RdbLoaderBase::ReadGeneric(int rdbtype) -> io::Result<OpaqueObj> {
+  bool is_string_type = RDB_TYPE_STRING == rdbtype;
   RdbVariant str_obj;
-  error_code ec = ReadStringObj(&str_obj);
+  error_code ec;
+  if (pending_read_.remaining) {
+    ec = ReadRemainingString(&str_obj);
+  } else {
+    ec = ReadStringObj(&str_obj, is_string_type);
+  }
   if (ec)
     return make_unexpected(ec);
 
-  if (StrLen(str_obj) == 0) {
+  if (!is_string_type && StrLen(str_obj) == 0) {
```

Comment:
if `is_string_type` is true, and `pending_read_.remaining`  is false we won't return `Unexpected` and we will miss that the rdb file is corrupted

Reply:
not sure I understood your comment correctly.
This logic is to enable empty string -> is_string_type=true && pending_read_.remaining is false and there for we do not return rdb file corrupted

Reply:
empty string is valid . you can add empty string to dragonfly.
set x ""

Reply:
oh that's true, Then if `pending_read_.remaining > 0` and `StrLen(str_obj) == 0)` then we have a corrupted rdb file since we know that the string is not an empty string (because we got remaining) but we just read 0!

Reply:
if remaining > 0 we enter ReadRemainingString and it will read from sink. if the length that I am trying to read is not available that it will return error

Reply:
gotcha cheers